funnel/views/project.py

```diff
@@ -332,21 +335,19 @@ def schedule_transition(self):
         return redirect(self.obj.url_for())
 
     @route('rsvp', methods=['POST'])
-    @render_with('rsvp.html.jinja2')
     @lastuser.requires_login
-    @requires_permission('view')
-    def rsvp(self):
-        form = RsvpForm()
+    def rsvp_transition(self):
+        rsvp = Rsvp.get_for(self.obj, current_auth.user, create=True, session_add=True)
```

Comment:
Don't create before validating the form. Use `rsvp = Rsvp.get_for(self.obj, current_auth.user, create=False)`